boards/arm/96b_wistrio/Kconfig.defconfig

```diff
@@ -11,4 +11,7 @@ config BOARD
 config REGULATOR
 	default y if LORA
 
+config I2C
+	default y if SENSOR
```

Comment:
This pattern is not ideal, but still, it's probably good enough for upstream boards and samples as it minimizes the maintenance work. It should probably be documented together with its limitations.

Reply:
> not sure if it is a good idea to default to y as long as SENSOR is enabled, what if user has an overlay to disable the i2c sensor in the default device tree, but has another spi sensor?

My understanding is that the default is should set to what you expect from a usual use case. These boards have sensors on the I2C bus, so this setting should be reasonable.
It may not always be a preferable default value in the complex situations given in the example.
But in such situations, I think it should be explicitly specify configurations, using with prj.conf for examply.